### PR TITLE
[ONNX] Refactor exporter errors

### LINKED ISSUE #135125: [ONNX] Merge the error classes

### PR DESCRIPTION
**Pull request by justinchuby**
Refactor exporter errors to combine old errors and new errors for API consistency.

This PR also

1. Removes the `_C._check_onnx_proto(proto)` call in the old exporter. We don't need the ONNX checker because it is limited.
2. Removes the `OnnxExporterError` defined in the dynamo module. This class unnecessarily stores the onnx program object, making it very bulky. Instead, we revert to use the plain OnnxExporterError defined in the `errors` module and use it as the base class for all errors.
3. Continues to expose `OnnxExporterError` in `torch.onnx` and the rest of the errors in `torch.onnx.errors`.
4. Removes the `CheckerError` and `InvalidExportOptionsError` from `torch.onnx`. This is BC breaking but should have low impact.
5. I did not rename existing errors out of compatibility considerations, even though `ExporterError` would have been more succinct. 

Fixes https://github.com/pytorch/pytorch/issues/135125

### CODE REVIEW

File: test/onnx/test_fx_to_onnx.py [hunk 1]
```diff
@@ -242,20 +242,6 @@ def forward(self, input):
...
             )
 
-        try:
-            torch.onnx.dynamo_export(
-                TraceModel(),
-                x,
-                export_options=torch.onnx.ExportOptions(onnx_registry=registry),
-            )
-        except torch.onnx.OnnxExporterError as e:
-            assert_has_diagnostics(
```
Comment (titaiwangms) [inline test/onnx/test_fx_to_onnx.py:252]: Maybe this should stay until sarif is deleted?

File: test/onnx/onnx_test_common.py [hunk 2]
```diff
@@ -286,35 +285,19 @@ def run_test_with_fx_to_onnx_exporter_and_onnx_runtime(
...
+                    dynamic_shapes=self.dynamic_shapes,
+                    diagnostic_options=torch.onnx.DiagnosticOptions(
+                        verbosity_level=logging.DEBUG
                     ),
-                )
-        except torch.onnx.OnnxExporterError as e:
-            export_error = e
-            onnx_program = e.onnx_program
-
-        if diagnostics.is_onnx_diagnostics_log_artifact_enabled():
```
Comment (titaiwangms) [inline test/onnx/onnx_test_common.py:307]: onnx_diagnostics still exists

Comment (justinchuby) [reply test/onnx/onnx_test_common.py:307]: the exception no longer stores the onnx program so this logic is no longer relavent. Also this is just dumping the sarif file we don't need.

File: test/onnx/test_fx_to_onnx.py (hunk 1 above)
Comment (justinchuby) [reply test/onnx/test_fx_to_onnx.py:252]: I removed because the OnnxExporterError contains the information being checked here.

**Review by titaiwangms — APPROVED** [review]

**Pull request closed without merging**